src/fprime/fbuild/interaction.py

```diff
@@ -216,3 +279,215 @@ def new_component(
+            print("'" + char + "' is not a valid character. Enter a new port name:")
+        else:
+            valid_name = True
+    short_description = input(
+        "Short Description [{}]: ".format(defaults["short_description"])
+    )
+    while not valid_dir:
+        dir_name = input("Directory Name [{}]: ".format(defaults["dir_name"]))
+        char = is_valid_name(dir_name)
+        if char != "valid":
+            print(
+                "'" + char + "' is not a valid character. Enter a new directory name:"
+            )
+        else:
+            valid_dir = True
+
+    while not valid_namespace:
+        namespace = input("Port Namespace [{}]: ".format(defaults["namespace"]))
+        char = is_valid_name(namespace)
+        if char != "valid":
+            print(
+                "'" + char + "' is not a valid character. Enter a new directory name:"
+            )
+        else:
+            valid_namespace = True
+    while not args_done:
+        if arg_list == []:
+            add_arg = confirm("Would you like to add an argument?: ")
+        else:
+            add_arg = confirm("Would you like to add another argument?: ")
+        if add_arg:
+            arg_name = input("Argument name: ")
+            arg_type = input("Argument type: ")
+            arg_description = input("Short description of argument: ")
+            arg_list.append((arg_name, arg_type, arg_description))
+        else:
+            args_done = True
+    values = {
+        "port_name": port_name,
+        "short_description": short_description,
+        "dir_name": dir_name,
+        "namespace": namespace,
+        "arg_list": arg_list,
+    }
+
+    # Fill in blank values with defaults
+    for key in values:
+        if values[key] == "":
+            values[key] = defaults[key]
+    return values
+
+
+def make_namespace(deployment, cwd):
+    # Form the namespace from the path to the deployment
+    namespace_path = cwd.relative_to(deployment)
+    deployment_dir = deployment.name
+    whole_path = Path(deployment_dir, namespace_path)
+    namespace = whole_path.parent
+    namespace_formatted = str(namespace).replace(os.path.sep, "::")
+    return namespace_formatted
+
+
+def new_port(cwd: Path, deployment: Path, build: Build):
+    """ Uses cookiecutter for making new ports """
+    try:
+        print("[WARNING] **** fprime-util new is prototype functionality ****")
+        proj_root = build.get_settings("project_root", None)
+        if proj_root is not None:
+            proj_root = Path(proj_root)
+            proj_root_found = True
+        else:
+            proj_root_found = False
+
+        PATH = os.path.dirname(os.path.abspath(__file__))
+        TEMPLATE_ENVIRONMENT = Environment(
+            autoescape=False,
+            loader=FileSystemLoader(os.path.join(PATH, "../cookiecutter_templates")),
+            trim_blocks=False,
+        )
+        params = get_port_input(deployment.name)
+
+        context = {
```

Comment:
Isn't this context dict identical to the params dictionary?

Reply:
Whoops, you're right!